packit/distgit.py

```diff
@@ -212,10 +211,30 @@ def upload_to_lookaside_cache(self, archive_path: str) -> None:
         try:
             f.new_sources(sources=archive_path)
         except Exception as ex:
-            logger.error(f"`fedpkg new-sources` failed for some reason. "
-                         f"Either Fedora kerberos is invalid or there could be network outage.")
+            logger.error(
+                f"`fedpkg new-sources` failed for some reason. "
+                f"Either Fedora kerberos is invalid or there could be network outage."
+            )
             raise PackitException(ex)
 
+    def is_archive_on_lookaside_cache(self, archive_path: str) -> bool:
+        archive_name = os.path.basename(archive_path)
+        try:
+            res = requests.head(
+                f"https://src.fedoraproject.org/lookaside/pkgs/{self.package_name}/{archive_name}"
+            )
+            if res.ok:
+                logger.info(
+                    f"Archive {archive_name} found on lookaside cache (skipping upload)."
+                )
+                return True
+            logger.debug(f"Archive {archive_name} not found on the lookaside cache.")
+        except requests.exceptions.BaseHTTPError as ex:
+            logger.warning(
```

Comment:
this is not a warning to be honest: we are just doing cache hit/miss - if the archive is not there, it really just means we need to upload it, so this should be debug

Reply:
No, on the cache miss, you get `404` and  `res.ok == False` and then we are using `logger.debug`.

Reply:
oh! good point, I haven't realized that, so this should likely indeed be a warning